src/test/java/wolox/training/models/BookTest.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    @Test
+    public void whenSaveBook_thenReturnBook(){
+        //Given
+        Book bookTest = new Book(null,"Mystery","Paolo", "image.png", "Sacred Stone", "-", "Salamander","2000", 250,"123456789");
+        testEntityManager.persist(bookTest);
+        testEntityManager.flush();
+
+        //when
+        Book bookSaved = bookRepository.save(bookTest);
+
+        //then
+        assertEquals(bookSaved.getGenre(),bookTest.getGenre());
+        assertEquals(bookSaved.getPublisher(),bookTest.getPublisher());
+        System.out.println("whenSaveBook passed");
+    }
+
+    @Test
+    public void whenSaveBookWithGenreNull_thenReturnBook(){
+        //Given
+        Book bookTest = new Book(null,null,"Paolo", "image.png", "Sacred Stone", "-", "Salamander","2000", 250,"123456789");
+        testEntityManager.persist(bookTest);
+        testEntityManager.flush();
+
+        //when
+        Book bookSaved = bookRepository.save(bookTest);
+
+        //then
+        assertEquals(bookSaved.getAuthor(),bookTest.getAuthor());
+        assertEquals(bookSaved.getTitle(),bookTest.getTitle());
+        System.out.println("whenSaveBookWithGenreNull passed");
+    }
+
+    @Test
+    public void whenFindByAuthor_thenReturnBook(){
+        //Given
+        Book bookTest = new Book(null,"Mystery","Paolo", "image.png", "Sacred Stone", "-", "Salamander","2000", 250,"123456789");
+        testEntityManager.persist(bookTest);
+        testEntityManager.flush();
+
+        //when
+        Book bookFound = bookRepository.findFirstByAuthor(bookTest.getAuthor()).get();
+
+        //then
+        assertEquals(bookFound.getAuthor(),bookTest.getAuthor());
+        System.out.println("whenFinByAuthor passed");
+    }
+
+
+    @Test()
+    public void whenSaveBookWithTitleNull_thenThrowException() throws ConstraintViolationException {
+        //Given
+        Book bookTest = new Book(null,"Mystery","David", "image.png", null, "-", "Salamander","2000", 250,"123456789");
+
+        //when
+        ConstraintViolationException exception = assertThrows(ConstraintViolationException.class, () -> {
+            bookRepository.save(bookTest);
+            testEntityManager.flush();
+        });
+
+        //then
+        assertTrue(exception instanceof ConstraintViolationException);
+        System.out.println("whenSaveBookWithTitleNull passed");
+    }
+
+    @Test()
+    public void whenSaveBookWithIsbnNull_thenThrowException() throws ConstraintViolationException {
+        //Given
+        Book bookTest = new Book(null,"Mystery","David", "image.png", "title", "-", "Salamander","2000", 250,null);
+
+        //when
+        ConstraintViolationException exception = assertThrows(ConstraintViolationException.class, () -> {
+            bookRepository.save(bookTest);
+            testEntityManager.flush();
+        });
+        //then
+        assertTrue(exception instanceof ConstraintViolationException);
+        System.out.println("whenSaveBookWithIsbnNull passed");
```

Comment:
This print is not necessary, as it is a test, the result is already shown by the console